### PR TITLE
Introducing ImmutableSetSerializer

### PR DESCRIPTION
**Pull request by fgaule**
Hi folks, 

My team has developed some guava serializers. Here is one of them. I would like you to check it out and tell me if we are ok to pull-request the others.

Cheers!

### CODE REVIEW

File: src/main/java/de/javakaffee/kryoserializers/guava/ImmutableSetSerializer.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        output.writeInt(object.size(), true);
+        for (Object elm : object) {
+            kryo.writeClassAndObject(output, elm);
+        }
+    }
+
+    @Override
+    public ImmutableSet<Object> read(Kryo kryo, Input input, Class<ImmutableSet<Object>> type) {
+        final int size = input.readInt(true);
+        final Object[] set = new Object[size];
```
Comment (magro) [inline src/main/java/de/javakaffee/kryoserializers/guava/ImmutableSetSerializer.java]: You could also use ImmutableSet.Builder, might be more efficient.

Comment (fgaule) [inline src/main/java/de/javakaffee/kryoserializers/guava/ImmutableSetSerializer.java]: We develop like this cause ImmutableListSerializer is using #copyOf, so we assume someone benchmark it against builder. 
Do you think we should move both implementations to use builders?
Your call!

Comment (magro) [inline src/main/java/de/javakaffee/kryoserializers/guava/ImmutableSetSerializer.java]: ImmutableListSerializer came in via a pull request as well. 

Looking at the source, ImmutableSet builder would be a little bit more efficient for less than five elements (the array.clone as done by copyOf is saved), for more elements copyOf seems to get more efficient. So we can use the copyOf... I'll merge and make a release soon.

**Comment by magro** [comment]
One thing's left: can you please update the readme to mention the new serializer? Thanks!

**Comment by fgaule** [comment]
There you go!. 

I will try to upload ImmutableMultimapSerializer and ImmutableMultimapSerializer as soon as i can

Thanks!

**Comment by magro** [comment]
Great, do you want to add them to this pull request or would you prefer do get a release with the Set serializer now?

**Comment by fgaule** [comment]
I would like to upload them in other PR. You could do a release with all
together after they get your approval. What do you think?

**Comment by magro** [comment]
Sounds good, so I'll wait with the release.

**Comment by magro** [comment]
Thanks for the PR again, it's on its way to maven central with release 0.32.

**Pull request merged**